modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/RowHandler.java

```diff
@@ -44,6 +45,23 @@ public interface RowHandler<RowT> {
     /** Concatenate two rows. */
     RowT concat(RowT left, RowT right);
 
+    /**
+     * Creates a new row containing only the fields specified in the provided mapping.
+     *
+     * <p>For example:
+     * <pre>
+     *    source row [5, 6, 7, 8] apply mapping [0, 2] with offset 1
+     *    result mapping will be [1, 3]
+     *    result row will be [6, 8]
+     * </pre>
+     *
+     * @param row Source row.
+     * @param mapping Target field indexes.
+     * @param offset Target field indexes offset.
+     * @return A new row with fields from the specified mapping.
+     */
+    RowT map(RowT row, int[] mapping, int offset);
```

Comment:
offset removed from the signature.

> why do we even need this method in the first place?
It seemed to me that depending on the implementation it can be implemented more efficiently.

> it does make sense to create new row explicitly
I didn't find how to get `rowschema` :pensive: to create a new row